### PR TITLE
MAINT: changed np.generic arguments to positional-only

### PR DESCRIPTION
**Pull request by BvB93**
Implementation of https://github.com/numpy/numpy/issues/16577:
* The arguments of `np.generic` constructor (more specifically its subclasses) are now marked as positional-only.

### CODE REVIEW

File: numpy/__init__.pyi [hunk 1]
```diff
@@ -479,24 +479,24 @@ class complex128(complexfloating):
...
 class character(_real_generic): ...  # type: ignore
 
 class bytes_(character):
     @overload
-    def __init__(self, value: object = ...) -> None: ...
+    def __init__(self, __value: object = ...) -> None: ...
     @overload
     def __init__(
-        self, value: object, encoding: str = ..., errors: str = ...
+        self, __value: object, encoding: str = ..., errors: str = ...
```
Comment (BvB93) [inline numpy/__init__.pyi:491]: The `encoding` and `errors` parameters in `np.bytes_` and `np.str_` are pretty much the only case where one *can* successfully supply a keyword argument.

Example
---------
``` python
>>> import numpy as np

>>> np.bytes_('bob', 'utf-8')
b'bob'

>>> np.bytes_('bob', 'utf-16')
b'\xff\xfeb\x00o\x00b'

>>> np.bytes_('bob', encoding='utf-16')
b'\xff\xfeb\x00o\x00b'
```

File: numpy/__init__.pyi [hunk 2]
```diff
@@ -382,38 +382,38 @@ class _real_generic(generic):  # type: ignore
...
 
 class object_(generic):
-    def __init__(self, value: object = ...) -> None: ...
+    def __init__(self, __value: object = ...) -> None: ...
 
 class datetime64:
     @overload
     def __init__(
-        self, _data: Union[datetime64, str, dt.datetime] = ..., _format: str = ...
+        self, __value: Union[datetime64, str, dt.datetime] = ..., __format: str = ...
```
Comment (eric-wieser) [inline numpy/__init__.pyi:393]: Is it possible to use positional-only argument syntax here instead, as

```suggestion
        self,value: Union[datetime64, str, dt.datetime] = ..., format: str = ..., /
```

Comment (BvB93) [reply numpy/__init__.pyi:393]: > I can't tell if this syntax is allowed in pyi files prior to python 3.8,

Unfortunately it isn't; mypy and the likes will start raising all sorts of alarm bells with such syntaxes.

**Comment by BvB93** [comment]
So the windows 32 bit test seems to get stuck on downloading something by the name of `mingw 32 bit`.
I think the error is unrelated to the changes introduced in this particular pull request?

**Comment by seberg** [comment]
Yes, don't worry about that test. We already tried reduce the number of mingw tests because of this, but it seems its still happening...

**Comment by person142** [comment]
Rerunning the azure test made it pass, but the one remaining Travis failure actually looks suspicious:

https://travis-ci.org/github/numpy/numpy/jobs/697614567

It seems like it might be hanging on `test_typing`?

**Comment by BvB93** [comment]
> It seems like it might be hanging on `test_typing`?

Odd, I'm not sure what initialized caused `test_typing` to get stuck, but restarting seems to have fixed the issue.

**Comment by person142** [comment]
Weird. Anywho, merged now, thanks @BvB93.

**Pull request merged**